file.go

```diff
@@ -87,7 +88,11 @@ func MakeOpenFile(h syscall.Handle) (io.ReadWriteCloser, error) {
 
 // closeHandle closes the resources associated with a Win32 handle
 func (f *win32File) closeHandle() {
-	if !f.closing {
+	f.closingLock.Lock()
+	alreadyClosing := f.closing
+	f.closing = true
+	f.closingLock.Unlock()
+	if !alreadyClosing {
```

Comment:
If `alreadyClosing` is true, that implies that `closeHandle()` was called from another goroutine.  This is the same behavior as today, since the existing code sets `f.closing = true` before waiting.

Does `closeHandle()` need to wait when called from multiple goroutines concurrently (does each call to `closeHandle()` need to wait)?